src/Simplify.cpp

```diff
@@ -525,7 +538,11 @@ class Simplify : public IRMutator {
         const Select *select_a = a.as<Select>();
         const Select *select_b = b.as<Select>();
 
-        if (const_int(a, &ia) &&
+        if (is_error_call(call_a)) {
```

Comment:
I don't think these are right. It's maybe right for undefined/indeterminate values, but signed overflow should not do this. Signed overflow should cancel too: if x is signed_integer_overflow(y), then x/x = 1, and x - x = 0.

In hindsight of this, my suggestion to just generalize signed_integer_overflow was clearly wrong. We need separate poison values for signed_integer_overflow, and indeterminate_value (or whatever), so different simplifier rules can handle them differently.